lib/resty/etcd.lua

```diff
@@ -46,30 +99,28 @@ function _M.new(opts)
         return nil, 'opts must be table'
     end
 
+    local protocol = opts and opts.protocol or "v3"
+    if protocol ~= "v3" then
+        return nil, 'only support etcd v3 api'
+    end
+
     opts.timeout = opts.timeout or 5    -- 5 sec
     opts.http_host = opts.http_host or "http://127.0.0.1:2379"
     opts.ttl  = opts.ttl or -1
 
-    local protocol = opts and opts.protocol or "v2"
     local serializer_name = typeof.string(opts.serializer) and opts.serializer
     opts.serializer = require_serializer(serializer_name)
 
-    if protocol == "v3" then
-        -- if opts special the api_prefix,no need to check version
-        if not opts.api_prefix or not utils.has_value(prefix_v3, opts.api_prefix) then
-            local ver, err = etcd_version(opts)
-            if not ver then
-                return nil, err
-            end
-            local sub_ver = ver.etcdserver:sub(1, 4)
-            opts.api_prefix = prefix_v3[sub_ver] or "/v3beta"
+    if not opts.api_prefix or not utils.has_value(prefix_v3, opts.api_prefix) then
+        local ver, err = etcd_version(opts)
```

Comment:
We don't need to detect the version. The only supported prefix now is `/v3`, see https://github.com/api7/lua-resty-etcd/blob/70b648fbfb7a70f6bb257a41350fbc780a7ff58f/lib/resty/etcd.lua#L9